ShareX.UploadersLib/FileUploaders/Uplea.cs

```diff
@@ -0,0 +1,231 @@
+        public string UpleaApiKey { get; set; }
+
+        public UpleaGetUserInformationResponse GetUserInformation(string apiKey)
+        {
+            var upleaGetUserInformationResponse = JsonConvert.DeserializeObject<UpleaGetUserInformationResponse>(GetUpleaResponse<UpleaGetUserInformationRequest>(new UpleaGetUserInformationRequest() { ApiKey = apiKey }));
+
+            if (upleaGetUserInformationResponse.Status)
+            {
+                return upleaGetUserInformationResponse;
+            }
+
+            return new UpleaGetUserInformationResponse();
+        }
+
+        private UpleaNode GetBestNode()
+        {
+            var getBestNodeResponse = JsonConvert.DeserializeObject<UpleaGetBestNodeResponse>(SendRequest(HttpMethod.POST, upleaBaseUrl + "get-best-node"));
+            return new UpleaNode(getBestNodeResponse.Result.Name, getBestNodeResponse.Result.Token);
+        }
+
+        public string GetApiKey(string username, string password)
+        {
+            var upleaGetApiKeyResponseStr = GetUpleaResponse(new UpleaGetApiKeyRequest() { Username = username, Password = password });
+
+            if (!string.IsNullOrEmpty(upleaGetApiKeyResponseStr))
+            {
+                try
+                {
+                    var upleaGetApiKeyResponse = JsonConvert.DeserializeObject<UpleaGetApiKeyResponse>(upleaGetApiKeyResponseStr);
+
+                    if (upleaGetApiKeyResponse.Status)
+                    {
+                        return upleaGetApiKeyResponse.Result.ApiKey;
+                    }
+                }
+                catch (JsonSerializationException ex)
+                {
+                    // For some reason the Uplea API is supposed to return a single object in the result property of the response, but when 
+                    // there is an error it returns an empty array. This is causing deserialziation to fail. Do we want to just query the JSON 
+                    // status before trying to deserialize? 
+
+                    System.Diagnostics.Debug.WriteLine("Deserialization of UpleaGetApiKeyResponse failed: {0}", ex.Message);
+
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private string GetUpleaResponse<T>(T upleaRequest) where T : IUpleaRequest
+        {
+            Dictionary<string, string> parameters = new Dictionary<string, string>();
+            parameters.Add("json", JsonConvert.SerializeObject(upleaRequest));
+            return SendRequestURLEncoded(upleaBaseUrl + upleaRequest.RequestType, parameters);
+        }
+
+        public override UploadResult Upload(Stream stream, string fileName)
+        {
+            var upleaBestNode = GetBestNode();
+
+            Dictionary<string, string> args = new Dictionary<string, string>();
+            args.Add("api_key", UpleaApiKey);
+            args.Add("token", upleaBestNode.Token);
+            args.Add("file_id[]", string.Format("{0}", Guid.NewGuid()));
```

Comment:
Previously was including more than just the Guid. Also, don't have to call ToString on NewGuid when using string.Format. I've removed string.Format here now.